sympy/polys/matrices/ddm.py

```diff
@@ -92,6 +92,22 @@ def __init__(self, rowslist, shape, domain):
         if not (len(self) == m and all(len(row) == n for row in self)):
             raise DDMBadInputError("Inconsistent row-list/shape")
 
+    def getitem(self, i, j):
+        return self[i][j]
+
+    def extract_slice(self, slice1, slice2):
+        m, n = self.shape
+        ri = range(m)[slice1]
+        ci = range(n)[slice2]
+        slice1 = slice(ri.start, ri.stop, ri.step)
+        slice2 = slice(ci.start, ci.stop, ci.step)
```

Comment:
I don't understand why we need to make new slices here. Why can't slice1 and slice2 be used directly?

Reply:
Oh sorry, it works without doing this, I forgot this is `list`, not `dict`.
Thanks.